src/napari/_vispy/canvas.py

```diff
@@ -779,3 +852,70 @@ def screenshot(self) -> QImage:
     def enable_dims_play(self, *args) -> None:
         """Enable playing of animation. False if awaiting a draw event"""
         self.viewer.dims._play_ready = True
+
+    def _update_scenegraph(self, event=None):
+        for camera in self.grid_cameras:
+            camera._2D_camera.parent = None
+            camera._3D_camera.parent = None
+            self._scene_canvas.events.draw.disconnect(camera.on_draw)
+        self.grid_cameras.clear()
+        self.grid_views.clear()
+
+        # grid are really not designed to be reset, so it's easier to replace it
+        self.grid.parent = None
+        self.grid = self.central_widget.add_grid(
+            border_width=0, spacing=self.viewer.grid.spacing
+        )
+
+        if self.viewer.grid.enabled:
+            self._setup_layer_views_in_grid()
+        else:
+            self._setup_single_view()
+
+        self._reorder_layers()
+        self._update_viewer_overlays()
+        self._on_interactive()
+
+    def _setup_single_view(self):
+        for napari_layer, vispy_layer in self.layer_to_visual.items():
+            vispy_layer.node.parent = self.view.scene
+            self._update_layer_overlays(napari_layer)
+
+    def _setup_layer_views_in_grid(self):
+        for napari_layer, vispy_layer in self.layer_to_visual.items():
+            row, col = self.viewer.grid.position(
+                self.viewer.layers.index(napari_layer),
+                len(self.viewer.layers),
+            )
+            view = self.grid[row, col]
+            view.border_width = self.viewer.grid.border_width
+            camera = VispyCamera(view, self.viewer.camera, self.viewer.dims)
+            self._scene_canvas.events.draw.connect(camera.on_draw)
+            self.grid_views.append(view)
+            self.grid_cameras.append(camera)
+
+            vispy_layer.node.parent = view.scene
+            self._update_layer_overlays(napari_layer)
+
+        self._highlight_selected_grid()
+
+    def _highlight_selected_grid(self):
+        if not self.viewer.grid.enabled:
+            return
+        for viewbox in self.grid_views:
+            viewbox.border_color = 'black'
```

Comment:
This fixes the border issue, see https://github.com/vispy/vispy/issues/1492
It looks like there is support to fix this on the VisPy side, so do whatever you think is best. 

```suggestion
    # any border_color != None will add a padding of +1, so we return if we want no border https://github.com/vispy/vispy/issues/1492
    if self.viewer.grid.border_width < 1:
        for viewbox in self.grid_views:
            viewbox.border_color=None # reset border_color to None as failsafe
        return
```

Reply:
Woah! Nice find @TimMonko :muscle:

Reply:
hehe nice find indeed!